gno.land/pkg/integration/gnoland.go

```diff
@@ -0,0 +1,367 @@
+package integration
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	vmm "github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+	"github.com/gnolang/gno/tm2/pkg/bft/config"
+	"github.com/gnolang/gno/tm2/pkg/bft/node"
+	"github.com/gnolang/gno/tm2/pkg/bft/privval"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/db"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+type IntegrationConfig struct {
+	SkipFailingGenesisTxs bool
+	SkipStart             bool
+	GenesisBalancesFile   string
+	GenesisTxsFile        string
+	ChainID               string
+	GenesisRemote         string
+	RootDir               string
+	GenesisMaxVMCycles    int64
+	Config                string
+}
+
+// NOTE: this is a copy of gnoland actual flags
+// XXX: a lot this make no sense for integration
+func (c *IntegrationConfig) RegisterFlags(fs *flag.FlagSet) {
+	fs.BoolVar(
+		&c.SkipFailingGenesisTxs,
+		"skip-failing-genesis-txs",
+		false,
+		"don't panic when replaying invalid genesis txs",
+	)
+	fs.BoolVar(
+		&c.SkipStart,
+		"skip-start",
+		false,
+		"quit after initialization, don't start the node",
+	)
+
+	fs.StringVar(
+		&c.GenesisBalancesFile,
+		"genesis-balances-file",
+		"./genesis/genesis_balances.txt",
+		"initial distribution file",
+	)
+
+	fs.StringVar(
+		&c.GenesisTxsFile,
+		"genesis-txs-file",
+		"./genesis/genesis_txs.txt",
+		"initial txs to replay",
+	)
+
+	fs.StringVar(
+		&c.ChainID,
+		"chainid",
+		"dev",
+		"the ID of the chain",
+	)
+
+	fs.StringVar(
+		&c.RootDir,
+		"root-dir",
+		"testdir",
+		"directory for config and data",
+	)
+
+	fs.StringVar(
+		&c.GenesisRemote,
+		"genesis-remote",
+		"localhost:26657",
+		"replacement for '%%REMOTE%%' in genesis",
+	)
+
+	fs.Int64Var(
+		&c.GenesisMaxVMCycles,
+		"genesis-max-vm-cycles",
+		10_000_000,
+		"set maximum allowed vm cycles per operation. Zero means no limit.",
+	)
+
+	fs.StringVar(
+		&c.Config,
+		"config",
```

Comment:
```suggestion
		"node-config",
```

Because we can have a -config flag and use this Register helper for a sub command: overlap.

Reply:
Those flags are from the original gnoland cmd, I think that changing it here might be confusing ?

Reply:
i've removed the flag since it's useless for integration command